simde/x86/ssse3.h

```diff
@@ -33,6 +33,158 @@ HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
 SIMDE_BEGIN_DECLS_
 
+SIMDE_FUNCTION_ATTRIBUTES
+simde__m128i
+simde_x_mm_deinterleaveeven_epi16 (simde__m128i a, simde__m128i b) {
+  simde__m128i_private
+    r_,
+    a_ = simde__m128i_to_private(a),
+    b_ = simde__m128i_to_private(b);
+  #if defined(SIMDE_ASSUME_VECTORIZATION) && defined(SIMDE_SHUFFLE_VECTOR_)
```

Comment:
We don't need the `SIMDE_ASSUME_VECTORIZATION` condition here.  If you've called this function you've already made a determination about your implementation strategy, and shuffling isn't going to perform any worse than the loop here (it will likely perform much better).